Docs: Guide to onboard Terramate to an existing Terraform project (#155)
terramate-io/terramate-docs: cli/on-boarding/terraform.md

# Get started with Terramate and Terraform

> In this guide, You will learn how to add Terramate to an existing Terraform project without changing any configuration.

Reviewer: ```suggestion
This guide will teach you how to onboard Terramate to any existing Terraform repository in less than 5 minutes.

Here's what you will learn:
- Install Terramate CLI
- Onboard Terramate CLI to any Terraform repository
- Sign up for a free Terramate Cloud account
- Sync your stacks to Terramate Cloud
```

## 1: Install Terramate

>  Install the [Terramate cli](../installation.md).

Reviewer: @AnnuCode can we at least add `brew install terramate` here and in case a user needs other installation methods, can we refer to https://terramate.io/docs/cli/installation#install-the-cli

Author: yes, that would be good. Thanks!

## 2: Import Terraform root modules to Terramate

Mark your Terraform root modules as Terramate stacks:

```bash
terramate create --all-terraform
```

> This command creates a `stack.tm.hcl` file in every existing Terraform root module, enabling Terramate to consider the Terraform root modules as [stacks](../stacks/index.md). 

Reviewer: ```suggestion
This command creates a `stack.tm.hcl` file in every existing Terraform root module, enabling Terramate to consider the Terraform root modules as [stacks](../stacks/index.md). Those files are used to configure the metadata such as `name`, `description`, `tags` and optionally the orchestration behavior of a stack. 
```

## 1: Install Terramate

Install the Terramate cli:

 ::: code-group
```sh [macOS]
brew install terramate
```

```sh [Ubuntu & Debian]
# Add the Terramate repo to your sources
echo "deb [trusted=yes] https://repo.terramate.io/apt/ /" \
  | sudo tee /etc/apt/sources.list.d/terramate.list

apt update
apt install terramate
```

```sh [Fedora & CentOS]
# Add the Terramate repo to your sources
sudo tee /etc/yum.repos.d/terramate.repo <<EOF
[terramate]
name=Terramate Repository
baseurl=https://repo.terramate.io/yum/
enabled=1
gpgcheck=0
EOF

dnf install terramate
```

```txt [Windows]
Download the binary from
https://github.com/terramate-io/terramate/releases
```
:::

For other installation methods, please see the [installation page](../installation.md).

## 2: Import Terraform root modules to Terramate

Mark your Terraform root modules as Terramate stacks:

```bash
terramate create --all-terraform
```
This command creates a `stack.tm.hcl` file in every existing Terraform root module, enabling Terramate to consider the Terraform root modules as [stacks](../stacks/index.md). 

For example, an existing Terraform project repo with two root modules:
```bash
.

> ├── alice
> │   ├── main.tf
> │   └── terraform.tfstate
> ├── bob
> │   ├── main.tf
> │   └── terraform.tfstate

Reviewer: ```suggestion
├── alice
│   ├── main.tf
│   └── backend.tf
├── bob
│   ├── main.tf
│   └── backend.tf
```

Reviewer: Technically, the `terramate create --all-terraform` command doesn't look for a `terraform.tfstate` state file but for a state backend configuration in any file within a directory which is configured using the `backend {}` block

``` 
After running the command, a `stack.tm.hcl` file is created in both root modules marking them as Terramate stacks:

```bash
.

> ├── alice
> │   ├── main.tf
> │   ├── stack.tm.hcl
> │   └── terraform.tfstate
> ├── bob
> │   ├── main.tf
> │   ├── stack.tm.hcl
> │   └── terraform.tfstate

Reviewer: ```suggestion
├── alice
│   ├── main.tf
│   ├── stack.tm.hcl
│   └── backend.tf
├── bob
│   ├── main.tf
│   ├── stack.tm.hcl
│   └── backend.tf
```
